### ISSUE TITLE
Document or fix .cordova/config.json

### ISSUE THREAD

**Issue by nofxx** [phonegap/phonegap-cli#513]
There's no docs or mention to the file '.cordova/config.json'.

```json
  {"lib":
    {"www":
      {"id":"com.phonegap.hello-world",
       "version":"master",
       "uri":"https://github.com/phonegap/phonegap-app-hello-world/archivemaster.tar.gz",
       "link":false}
   },"phonegap":{"id": XXXXXX}
}
```

Newbie catch: if you start a project with `cordova create` instead of `phonegap`:
`[error] ENOENT, open '/home/nofxx/.../phonegap_project/.cordova/config.json'`

Another catch: If you remove or change the app on build it will miserably fail #122 

If it's meant to be a system/runtime file, it should be created accordingly.
Or document and create a example file for the user with what needs to go there.
(Do I need to keep that www: for instance?)

Related to #113

**Comment by mwbrooks** [comment]
Hi @nofxx,

Thanks for the suggestions.

The `.cordova/config.json` is generated by Cordova, but also not used by Cordova anymore. It's in a strange place with regards to Cordova's support.

Regardless, the PhoneGap CLI uses `.cordova/config.json` to store a few preferences and the main one is the PGB app ID. Based on your suggestions, I think we can improve the experience and remove some of the errors that you've referenced.

1. Create `.cordova/config.json` when it doesn't exist.
2. Remove and recreate the ID when the app is not found.

Also, the `lib` key is populated the Cordova tooling, yet it is never used. It should be safe to remove.

I'll add this as a task for next week!

**Comment by nofxx** [comment]
@mwbrooks thanks for the explanation, crystal clear now.

1 & 2 are exactly what I mean, but let me add a suggestion:
Use something like `.phonegap.json`, `.pgbuild`  so we could just `rm -r .cordova`.
Makes it more clear imho.

Thank you.

**Comment by askmatey** [comment]
So, what is the schema for the `cordova/config.json` file?

**Comment by si458** [comment]
is the any update on this at all? 
maybe away around it?
as im unable to build to the phonegap servers because of this error :(

**Comment by ilhanyumer** [comment]
@si458 It doesn't help. It leads to other errors. http://imgur.com/a/DHiEa

**Comment by syardumi** [comment]
I ran into the same problem. I also fixed it with { }. Is there any update?

**Comment by muneebtatar** [comment]
I was facing the same problem. Fixed it by creating .cordova folder and a config.json file inside it with {}

**Comment by mwbrooks** [comment]
Hi everyone, I believe commit 2c82089bb3fb4b2f18eba511a88d11b4e7444780 should help to address this issue. It's now available on npm as `phonegap@5.2.2`.

Thanks for your patients and help in pointing this out!
